dask/dataframe/multi.py

```diff
@@ -946,10 +946,7 @@ def merge_asof(
     if not is_dask_collection(right):
         right = from_pandas(right, npartitions=1)
     if right_on is not None:
-        if on:
-            right = right.set_index(right_on, sorted=True)
-        else:
-            right = right.set_index(right_on, drop=False, sorted=True)
+        right = right.set_index(right_on, drop=(left_on == right_on), sorted=True)
```

Comment:
What happens in the case where some of the columns are equal, but not all?

Reply:
It looks like only a single column is allowed for `on`/`left_on`/`right_on`, in which case this wouldn't be an issue.

Reply:
Yeah, these can only be single column names.